lib/Backend/Lower.cpp

```diff
@@ -25189,6 +25189,48 @@ Lowerer::LowerGetCachedFunc(IR::Instr *instr)
     return instrPrev;
 }
 
+IR::Instr *
+Lowerer::LowerCommitScope(IR::Instr *instrCommit)
+{
+    IR::Instr *instrPrev = instrCommit->m_prev;
+    IR::RegOpnd *baseOpnd = instrCommit->UnlinkSrc1()->AsRegOpnd();
+    IR::Opnd *opnd;
+    IR::Instr * insertInstr = instrCommit->m_next;
+
+    // Write undef to all the local var slots.
+
+    opnd = IR::IndirOpnd::New(baseOpnd, Js::ActivationObjectEx::GetOffsetOfCommitFlag(), TyInt8, this->m_func);
+    instrCommit->SetDst(opnd);
+    instrCommit->SetSrc1(IR::IntConstOpnd::New(1, TyInt8, this->m_func));
+    LowererMD::ChangeToAssign(instrCommit);
+
+    const Js::PropertyIdArray *propIds = instrCommit->m_func->GetJITFunctionBody()->GetFormalsPropIdArray();
+
+    uint firstVarSlot = (uint)Js::ActivationObjectEx::GetFirstVarSlot(propIds);
+    if (firstVarSlot < propIds->count)
+    {
+        // On ARM, instead of re-using the address of "undefined" for each store, put the address in a register
+        // and re-use that. (Would that be good for x86/amd64 as well?)
```

Comment:
nit: This comment seems to be very out-of-date, seeing as we were doing this on x86/amd64 already; consider removing it, or replacing it with one that documents the behavior.